### ISSUE TITLE
Using Python from Rust - Basic example in README fails

### ISSUE THREAD

**Issue by memark** [PyO3/pyo3#4466]
### Bug Description

Trying to run the first basic example in the section Using Python from Rust
https://github.com/PyO3/pyo3?tab=readme-ov-file#using-python-from-rust fails.

### Steps to Reproduce

Do what the README says here https://github.com/PyO3/pyo3?tab=readme-ov-file#using-python-from-rust
1. cargo new
2. add pyo3 0.22.2 to Cargo.toml
3. paste the basic example
4. cargo run

### Backtrace

```shell
cargo run
   Compiling once_cell v1.19.0
   Compiling unindent v0.2.3
   Compiling cfg-if v1.0.0
   Compiling libc v0.2.158
   Compiling pyo3-build-config v0.22.2
   Compiling memoffset v0.9.1
   Compiling pyo3-ffi v0.22.2
   Compiling pyo3-macros-backend v0.22.2
   Compiling pyo3 v0.22.2
   Compiling pyo3-macros v0.22.2
   Compiling pyo3-test v0.1.0 (/private/var/folders/pz/23v9mn0s78scfwbxw1sb5zqc0000gn/T/tmp.yGGe8i4W4l/pyo3-test)
warning: unused import: `pyo3::types::IntoPyDict`
 --> src/main.rs:3:5
  |
3 | use pyo3::types::IntoPyDict;
  |     ^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0599]: no method named `import` found for struct `pyo3::Python` in the current scope
 --> src/main.rs:7:22
  |
7 |         let sys = py.import("sys")?;
  |                      ^^^^^^
  |
help: there is a method `import_bound` with a similar name
  |
7 |         let sys = py.import_bound("sys")?;
  |                      ~~~~~~~~~~~~

error[E0599]: no method named `import` found for struct `pyo3::Python` in the current scope
  --> src/main.rs:10:33
   |
10 |         let locals = [("os", py.import("os")?)].into_py_dict(py);
   |                                 ^^^^^^
   |
help: there is a method `import_bound` with a similar name
   |
10 |         let locals = [("os", py.import_bound("os")?)].into_py_dict(py);
   |                                 ~~~~~~~~~~~~

error[E0599]: no method named `eval` found for struct `pyo3::Python` in the current scope
  --> src/main.rs:12:31
   |
12 |         let user: String = py.eval(code, None, Some(&locals))?.extract()?;
   |                               ^^^^ method not found in `Python<'_>`

For more information about this error, try `rustc --explain E0599`.
warning: `pyo3-test` (bin "pyo3-test") generated 1 warning
error: could not compile `pyo3-test` (bin "pyo3-test") due to 3 previous errors; 1 warning emitted
```

### Your operating system and version

MacOS 14.6.1

### Your Python version (`python --version`)

Python 3.11.8

### Your Rust version (`rustc --version`)

rustc 1.80.1 (3f5fd8dd4 2024-08-06)

### Your PyO3 version

0.22.2

### How did you install python? Did you use a virtualenv?

pyenv

### Additional Info

Maybe I'm missing something obvious here, but I would love to get this simple example to work.

**Comment by messense** [comment]
Note that the README in this repo reflects the main branch not the released version 0.22.2 you were using.

Please refer to https://pyo3.rs/ for user guide of released versions.

**Comment by memark** [comment]
@messense Thanks for your quick answer.

>Note that the README in this repo reflects the main branch not the released version 0.22.2 you were using.

I used version 0.22.2 because that's what the README told me to use. If that's incorrect, maybe the README could actually point to the main branch via `git:` or something similar?

>Please refer to https://pyo3.rs/ for user guide of released versions.

I tried this example and it works beautifully. Thanks!
